syncano/connection.py

```diff
@@ -236,9 +236,12 @@ def make_request(self, method_name, path, **kwargs):
         """
         data = kwargs.get('data', {})
 
-        files = {k: v for k, v in data.iteritems()
-                 if hasattr(v, 'read')}
-        map(data.pop, files.keys())
+        if method_name == 'POST':
+            files = data.pop('files', {})
+        else:
+            files = {k: v for k, v in data.iteritems()
+                     if hasattr(v, 'read')}
```

Comment:
Maybe here checks for read and write? It's always less possible to have interface with read and write than only read :) Dunno :)